platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineRegion.java

```diff
@@ -0,0 +1,241 @@
+     */
+
+    public long getID() {
+        return mId;
+    }
+
+    public OfflineRegionDefinition getDefinition() {
+        return mDefinition;
+    }
+
+    public OfflineRegionMetadata getMetadata() {
+        return mMetadata;
+    }
+
+    private Handler getHandler() {
+        if (handler == null) {
+            handler = new Handler(Looper.getMainLooper());
+        }
+
+        return handler;
+    }
+
+    /**
+     * Register an observer to be notified when the state of the region changes.
+     */
+    public void setObserver(@NonNull final OfflineRegionObserver observer) {
+        setOfflineRegionObserver(this, new OfflineRegionObserver() {
+            @Override
+            public void onStatusChanged(final OfflineRegionStatus status) {
+                getHandler().post(new Runnable() {
+                    @Override
+                    public void run() {
+                        observer.onStatusChanged(status);
+                    }
+                });
+            }
+
+            @Override
+            public void onError(final OfflineRegionError error) {
+                getHandler().post(new Runnable() {
+                    @Override
+                    public void run() {
+                        observer.onError(error);
+                    }
+                });
+            }
+        });
+    }
+
+    /**
+     * Pause or resume downloading of regional resources.
+     */
+    public void setDownloadState(@DownloadState int state) {
+        setOfflineRegionDownloadState(this, state);
+    }
+
+    /**
+     * Retrieve the current status of the region. The query will be executed
+     * asynchronously and the results passed to the given callback which will be
+     * executed on the main thread.
+     */
+    public void getStatus(@NonNull final OfflineRegionStatusCallback callback) {
+        getOfflineRegionStatus(this, new OfflineRegionStatusCallback() {
+            @Override
+            public void onStatus(final OfflineRegionStatus status) {
+                getHandler().post(new Runnable() {
+                    @Override
+                    public void run() {
+                        callback.onStatus(status);
+                    }
+                });
+            }
+
+            @Override
+            public void onError(final String error) {
+                getHandler().post(new Runnable() {
+                    @Override
+                    public void run() {
+                        callback.onError(error);
+                    }
+                });
+            }
+        });
+    }
+
+    /**
+     * Remove an offline region from the database and perform any resources evictions
+     * necessary as a result.
+     *
+     * Eviction works by removing the least-recently requested resources not also required
+     * by other regions, until the database shrinks below a certain size.
+     *
+     * Note that this method takes ownership of the input, reflecting the fact that once
```

Comment:
Removed in https://github.com/mapbox/mapbox-gl-native/commit/21b52c6d7c376d6e7e14e5b10fcdb40187c5f868.